ocs_ci/ocs/debug.py

```diff
@@ -0,0 +1,283 @@
+        else:
+            self.deployment_name = deployment_name
+
+        return True
+
+    def __get_data_path(self):
+        """
+        Get --data-path for deployment based on osd-id
+        """
+        data_path = self.data_path
+        for deployment in get_osd_deployments():
+            if deployment.name == f"{self.deployment_name}-debug":
+                osd_id = deployment.pods[0].labels["ceph-osd-id"]
+                data_path += f"ceph-{osd_id}"
+        return data_path
+
+    def __get_osd_debug_pod(self):
+        """
+        Get osd pod corresponding to the osd deployment
+        """
+        debug_pod = " "
+        for deployment in get_osd_deployments():
+            if deployment.name == f"{self.deployment_name}-debug":
+                debug_pod = deployment.pods[0]
+        return debug_pod
+
+    def run_cot_list_pgs(self, deployment_name):
+        """
+        Run COT list PG operation
+
+        Args:
+            deployment_name: Name of the original deployment thats in debug
+
+        Returns:
+            pgs: List of PGS
+        """
+        self.__validate_deployment(deployment_name)
+        data_path = self.__get_data_path()
+        command = self.cot_cmd + f" --data-path {data_path} --op list-pgs"
+        pgs = self.__get_osd_debug_pod().exec_cmd_on_pod(command=command).split(" ")
+        return pgs
+
+
+class MonStoreTool(RookCephPlugin):
+    """
+    This is to perform MonStoreTool related operations on Mon debug pod
+
+    """
+
+    def __init__(
+        self, deployment_name=None, store_path="/var/lib/ceph/mon/", *args, **kwargs
+    ):
+        super().__init__(*args, **kwargs)
+        self.store_path = store_path
+        self.mot_cmd = "ceph-monstore-tool"
+        self.deployment_name = deployment_name
+
+    def __validate_deployment(self, deployment_name):
+        """
+        Validate if the deployment is debug mode
+        before performing MonStoreTool operations
+        """
+        if not deployment_name:
+            if not self.deployment_name:
+                raise Exception(
+                    "Need to pass the deployment_name either when initialising "
+                    "the CephObjectStoreTool or while running the COT!!"
+                )
+        if deployment_name not in self.deployment_in_debug.keys():
+            raise Exception("Please put the osd deployment in debug mode first!")
+        else:
+            self.deployment_name = deployment_name
+
+        return True
+
+    def __get_store_path(self):
+        """
+        This returns the store-path for the corresponding
+        Mon id
+        """
+        store_path = self.store_path
+        for deployment in get_mon_deployments():
+            if deployment.name == f"{self.deployment_name}-debug":
+                mon_id = deployment.pods[0].labels["ceph_daemon_id"]
+                store_path += f"ceph-{mon_id}"
+        return store_path
+
+    def __get_mon_debug_pod(self):
```

Comment:
We can have one helper function with `deployment` passed to it `get_debug_pod(self, deployment)` and refactor `__get_mon_debug_pod` and `__get_osd_deployments`

Reply:
if we make them as single method then we would have write seperate code to figure out which deployment are we looking at.. hence we have seperate methods, one for mon deployments and one for osd deployments